kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -486,6 +479,30 @@ private long doCommit(
     }
   }
 
+  private void incrementMetricsForFileActionRow(Row fileActionRow, TransactionMetrics txnMetrics) {
+    txnMetrics.totalActionsCounter.increment();
+    if (!fileActionRow.isNullAt(ADD_FILE_ORDINAL)) {
+      txnMetrics.addFilesCounter.increment();
+      txnMetrics.addFilesSizeInBytesCounter.increment(
+          new AddFile(fileActionRow.getStruct(ADD_FILE_ORDINAL)).getSize());
+      // TODO increment fileSizeHistogram
+    } else if (!fileActionRow.isNullAt(REMOVE_FILE_ORDINAL)) {
+      txnMetrics.removeFilesCounter.increment();
+      RemoveFile removeFile = new RemoveFile(fileActionRow.getStruct(REMOVE_FILE_ORDINAL));
+      long fileSize =
+          removeFile
+              .getSize()
+              .orElseThrow(
+                  () ->
+                      new IllegalStateException(
```

Comment:
Question: is it worth creating our internal delta errors class? would that clean up the code?

Reply:
I'm not opposed to this but in what way does this clean up the code?

Reply:
It would be 1 line `DeltaErrorsXXX.yyy()` instead of 4

Reply:
Ah I see :) I was thinking error class like `KernelIllegalStateException` haha

Sure that makes sense to me

Reply:
Done